File: src/main/java/sk/hudak/prco/service/GroupProductKeywordsService.java
package sk.hudak.prco.service;

import sk.hudak.prco.dto.group.GroupProductKeywordsCreateDto;
import sk.hudak.prco.dto.group.GroupProductKeywordsFullDto;

import java.util.Optional;

public interface GroupProductKeywordsService {

    /**
     * @param groupProductKeywordsCreateDto data from creating of new keyword for given group
     * @return primary key id
     */
    Long createGroupProductKeywords(GroupProductKeywordsCreateDto groupProductKeywordsCreateDto);

    /**
     * @param groupId group id
     * @return
     */
    Optional<GroupProductKeywordsFullDto> getGroupProductKeywordsByGroupId(Long groupId);

    /**
     * @param groupId group id
     */
    void removeAllKeywordForGroupId(Long groupId);
}

File: src/main/java/sk/hudak/prco/service/GroupProductKeywordsService.kt
package sk.hudak.prco.service

import sk.hudak.prco.dto.GroupProductKeywordsCreateDto
import sk.hudak.prco.dto.GroupProductKeywordsFullDto

interface GroupProductKeywordsService {

    /**
     * @param createDto data from creating of new keyword for given group
     * @return primary key id
     */
    fun createGroupProductKeywords(createDto: GroupProductKeywordsCreateDto): Long

    /**
     * @param groupId group id
     * @return
     */
    fun getGroupProductKeywordsByGroupId(groupId: Long): GroupProductKeywordsFullDto?

    /**
     * @param groupId group id
     */
    fun removeAllKeywordForGroupId(groupId: Long)

    fun findAllGroupProductKeywords(): List<GroupProductKeywordsFullDto>
}
